src/pymatgen/analysis/diffraction/xrd.py

```diff
@@ -120,7 +120,8 @@ def __init__(self, wavelength="CuKa", symprec: float = 0, debye_waller_factors=N
             self.radiation = wavelength
             self.wavelength = WAVELENGTHS[wavelength]
         else:
-            raise TypeError(f"{type(wavelength)=} must be either float, int or str")
+            wavelength_type = type(wavelength).__name__
```

Comment:
Ah, thanks a lot for fixing all those (`type(var).__name__` for name directly instead of `<class 'xxx'>`). I totally forgot about them.